### PR TITLE
MOTECH-2839: Added Cohort Query Member Task Trigger

### PR DESCRIPTION
**Pull request by afijal**

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/OpenMRSTriggerBuilder.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.motechproject.openmrs.service.OpenMRSConfigService;
+import org.motechproject.openmrs.tasks.constants.DisplayNames;
+import org.motechproject.openmrs.tasks.constants.Keys;
+import org.motechproject.tasks.contract.EventParameterRequest;
+import org.motechproject.tasks.contract.TriggerEventRequest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OpenMRSTriggerBuilder {
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/OpenMRSTriggerBuilder.java:17]: add javadoc

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/constants/DisplayNames.java [hunk 2]
```diff
@@ -54,10 +54,13 @@
     public static final String STATE_START_DATE = "openMRS.program.state.startDate";
     public static final String PROGRAM_ENROLLMENT_ATTRIBUTES = "openMRS.programEnrollment.attributes";
 
+    //CohortQuery triggers
+    public static final String COHORTQUERY_UUID = "openMRS.cohortQuery.uuid";
+    public static final String PATIENT_DISPLAY = "openMRS.patient.display";
+    public static final String COHORT_GOTMEMBER = "Got member from cohortquery";
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/constants/DisplayNames.java]: property should be used here

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/constants/Keys.java [hunk 3]
```diff
@@ -55,6 +55,11 @@
     public static final String STATE_START_DATE = "startDate";
     public static final String PROGRAM_ENROLLMENT_ATTRIBUTES = "programEnrollmentAttributes";
 
+    //Cohortquery triggers
+    public static final String PATIENT_ID = "PatientId";
+    public static final String PATIENT_DISPLAY = "PatientDisplay";
+    public static final String COHORT_QUERY_ID = "CohortQueryId";
```
Comment (craigappl) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/constants/Keys.java]: The user should see "CohortQueryUUID" to keep in line with OpenMRS Reporting REST API definition

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/DisplayNameHelper.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Helper class for building display names for the OpenMRS task channel and data provider.
+ */
+public final class DisplayNameHelper {
+
+    /**
+     * Builds a display name in the form of "subject: name [config-name]". Example - Received Form: Birth [myConfig].
+     * If the name is blank, this will be formatted as "name [config-name]".
+     * @param subject the subject with which the display name should start
+     * @param name the name of the element (form, case, etc.), if blank will be omitted
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/DisplayNameHelper.java]: form, case, those types are from commcare not openmrs

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/OpenMRSTriggerBuilder.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import org.motechproject.openmrs.tasks.constants.DisplayNames;
+import org.motechproject.openmrs.tasks.constants.Keys;
+import org.motechproject.tasks.contract.EventParameterRequest;
+import org.motechproject.tasks.contract.TriggerEventRequest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The <code>OpenMRSTriggerBuilder</code> class builds case triggers for OpenMRS module.
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/OpenMRSTriggerBuilder.java]: 'case triggers' ? just triggers

File: openmrs/src/main/resources/webapp/messages/messages.properties [hunk 6]
```diff
@@ -151,3 +151,6 @@ openMRS.config.error.motechIDRequired=Motech ID is required.
 openMRS.config.error.patientIdentifierTypeExists=Patient Identifier type already in use.
 openMRS.config.info.motechID=The name of the OpenMRS identifier used by MOTECH. This must match the identifier that you will create in OpenMRS. The default is MOTECH Id.
 
+#Cohort query triggers
+openMRS.cohortQuery.uuid=CohortQuery UUID
+openMRS.cohortQuery.gotMember="Got member from cohortquery"
```
Comment (kmadej) [inline openmrs/src/main/resources/webapp/messages/messages.properties]: remove quotation, why 'got' not 'get' ? Additionally this trigger name sounds bad. In ticket description there is an information about 'CohortQuery Report Member' task trigger

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/constants/Keys.java [hunk 7]
```diff
@@ -55,6 +55,11 @@
     public static final String STATE_START_DATE = "startDate";
     public static final String PROGRAM_ENROLLMENT_ATTRIBUTES = "programEnrollmentAttributes";
 
+    //Cohortquery triggers
+    public static final String PATIENT_ID = "PatientId";
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/constants/Keys.java]: PATIENT_UUID - > PatientUuid

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/constants/Keys.java (hunk 3 above)
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/constants/Keys.java]: uuid

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/OpenMRSTriggerBuilder.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        List<TriggerEventRequest> triggers = new ArrayList<>();
+
+        List<EventParameterRequest> parameterRequests = new ArrayList<>();
+        parameterRequests.add(new EventParameterRequest(DisplayNames.PERSON_UUID, Keys.PATIENT_UUID));
+        parameterRequests.add(new EventParameterRequest(DisplayNames.PATIENT_DISPLAY, Keys.PATIENT_DISPLAY));
+        parameterRequests.add(new EventParameterRequest(DisplayNames.COHORTQUERY_UUID, Keys.COHORT_QUERY_UUID));
+
+        String displayName = DisplayNameHelper.buildDisplayName(DisplayNames.COHORT_REPORTMEMBER, configName);
+
+        triggers.add(new TriggerEventRequest(displayName, EventKeys.BASE_SUBJECT + "Cohort.ReportMember", null, parameterRequests));
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/OpenMRSTriggerBuilder.java]: use constant for event subject

**Comment by kmadej** [comment]
test this please

**Pull request merged**